### PR TITLE
Add Amiga, Switch, 3DS, and Vita definitions

### PR DESCRIPTION
**Pull request by glebm**
Adapted from the DevilutionX port, cleaned up.

Platform-specific credits:

* @MBeijer - Amiga
* @rsn8887 - Switch
* @MrHuu - 3DS
* @isage - Vita

/cc @AJenbo

### CODE REVIEW

**Comment by AJenbo** [comment]
Looks like this should make future ports a bit simpler :+1:

**Comment by ladislav-zezula** [comment]
Are all those new platforms Linux-compatible? If yes, then define STORMLIB_LINUX for them and it will work as well.

File: src/FileStream.cpp [hunk 1]
```diff
@@ -876,7 +858,7 @@ static void BaseHttp_Close(TFileStream * pStream)
     if(pStream->Base.Http.hInternet != NULL)
```
Comment (ladislav-zezula) [inline src/FileStream.cpp:858]: Given that those Internet functions are no longer used, I am afraid you are working on an old version of StormLib.

**Comment by ladislav-zezula** [comment]
I Moved the HAIKU platform under MAC (Aka, PLATFORM_MAC now also includes HAIKU).
Could you do the same with the other platforms as well?
https://github.com/ladislav-zezula/StormLib/blob/c8063655be321191533952792351188d8ccbab87/src/StormPort.h#L133

**Comment by ladislav-zezula** [comment]
I solved the multiple platform #ifdefs by making the new platforms Mac-compatible. For example, on the AMIGA platform, both `STORMLIB_AMIGA` and `STORMLIB_MAC` are defined. Therefore, the Mac compatible code will be used.

Thank you for the contribution!

File: src/FileStream.cpp (hunk 1 above)
Comment (glebm) [reply src/FileStream.cpp:858]: We do not use these functions either, I only added the comment here for consistency

**Pull request closed without merging**